highfidelity/hifi
# Cleanup entity dependencies

Our dependency tree is somewhat tangled currently.  For instance, this is the current dependency set for the `entities` library.

![image](https://user-images.githubusercontent.com/1533642/27404270-4b24bb04-5682-11e7-97f7-53134a5d71f8.png)

Since `entities` has nothing to do with rendering, many of these dependencies make no sense.  In particular the dependency on `animation` comes from a need to be able to parse animation metadata, and the dependency on `audio` comes because the entities can store a collisionSound, but while all entities need to be able to store information about their collisionSound, non-renderable entities never need to play the sound.

If we can clean some of this up, we can reduce the dependencies to 

![image](https://user-images.githubusercontent.com/1533642/27407095-fb5ce402-568b-11e7-82e1-525a17638f0c.png)

## Work done

* migrate AnimationDetails and AnimationLoop from `animation` to `shared`
* migrate `AnimationPointer _animation` from `ModelEntityItem` to `RenderableModelEntityItem`
* migrate `MeshFace` and `MeshProxy` to `shared/RegisteredMetaTypes.h`
  * Added `SimpleMeshProxy` to implement the required functionality in `entities-renderer`

## Testing

* Verify collision sounds still work 
* Verify animations still work for entities (dev-playa dancing robot)
* Verify `Model` scripting interface still works as expected
* Verify that from scripts the entity properties `naturalDimensions` and `naturalPosition` are still valid in places where they were before
